pex/cli/commands/lock.py

```diff
@@ -191,9 +196,32 @@ def add_extra_arguments(
 
     def _create(self):
         # type: () -> Result
-        lock_configuration = LockConfiguration(style=self.options.style)
+        target_configuration = target_options.configure(self.options)
+        if self.options.style == LockStyle.UNIVERSAL:
+            if target_configuration.pythons or target_configuration.platforms:
+                return Error(
+                    "When creating a {universal} lock, the interpreters the resulting lock applies "
+                    "to can only be constrained via --interpreter-constraint. There {were} "
+                    "{num_pythons} --python and {num_platforms} --platform specified.".format(
+                        universal=LockStyle.UNIVERSAL.value,
+                        were="were" if len(target_configuration.pythons) > 1 else "was",
+                        num_pythons=len(target_configuration.pythons),
+                        num_platforms=len(target_configuration.platforms),
+                    )
+                )
+            lock_configuration = LockConfiguration(
+                style=LockStyle.UNIVERSAL,
+                requires_python=tuple(
+                    str(interpreter_constraint.specifier)
```

Comment:
N.B.: Right now this is lossy on the project_name portion of the interpreter requirement; i.e.: for an explicit 'PyPy>=3' or 'CPython>=3' We'll resolve distributions not implied by the implementation constraint. I'd like to circle back to handle that: #1580